faucet/components/Faucet.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useEffect } from 'react'
+import { isAddress } from 'web3-utils'
+import type { Networks } from '@hoprnet/hopr-ethereum/scripts/addresses'
+
+async function mint(
+  network: Networks,
+  address: string
+): Promise<{
+  success: boolean
+  transactionHash?: string
+  message?: string
+}> {
+  const response = await fetch(`/api/mint?network=${network}&address=${address}`)
+
+  if (response.status === 200) {
+    const { transactionHash } = await response.json()
+    return {
+      success: true,
+      transactionHash,
+    }
+  } else {
+    return {
+      success: false,
+      message: response.statusText,
+    }
+  }
+}
+
+function Faucet({ network }: { network: Networks }) {
+  const [address, setAddress] = useState<string>(undefined)
+  const [isValidAddress, setIsValidAddress] = useState<boolean>(undefined)
+  const [status, setStatus] = useState<'PENDING' | 'SUCCESS' | 'FAILURE'>(undefined)
```

Comment:
Maybe we should create a `type` for this. Also, is it a standard to create the `state` as `undefined` for all types?

Reply:
I should document this better, I set `undefined` to signify that for example `address` was not set yet, I am not sure if this is the best way to do this though, I think adding some comments would certainly help